### ISSUE TITLE
[BUG] False positives for rule AZR-000279?

### ISSUE THREAD

**Issue by ThijmenDam** [Azure/PSRule.Rules.Azure#3407]
### Existing rule

Azure.Deployment.OutputSecretValue (AZR-000279)

### Description of the issue

Perhaps I am missing something, but it seems that AZR-000279 is incorrectly flagged in the following situation.

Our team implements the [AVM Storage Account](https://github.com/Azure/bicep-registry-modules/tree/main/avm/res/storage/storage-account) module like so:

```bicep
// function.data.bicep

module storageAccountFunction 'br/public:avm/res/storage/storage-account:0.20.0' = {
  name: XYZ
  params: {
    ....
  }
}

output storageAccountFunctionId string = storageAccountFunction.outputs.resourceId
```

This triggers AZR-000279:

```
 -> function.data.bicep : Microsoft.Resources/deployments [6/7]

    [FAIL] Azure.Deployment.OutputSecretValue (AZR-000279)
    | Template: function.data.bicep:0:0

    | RECOMMEND:
    | Consider removing any deployment output values that return secret values or use
    | secure outputs.

    | REASON:
    | - Path outputs.storageAccountFunctionId: The output 'storageAccountFunctionId' exposes a secure value.

    | HELP:
    | - https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Deployment.OutputSecretValue/
```

However, when I look at the AVM module's [source](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/storage/storage-account/main.bicep), we can see that the `resourceId` output is **not** a secure output:

```bicep
// br/public:avm/res/storage/storage-account:0.20.0 - main.bicep

....
@description('The resource ID of the deployed storage account.')
output resourceId string = storageAccount.id
....
```

Is this a false positive, or am I missing something here?

We can "resolve" the issue by adding the `@secure()` decorator to our module's output variable, but I do not see why this should be necessary considering that the AVM module's output is not marked as secure.

### Reproduction

1. Implement module `br/public:avm/res/storage/storage-account:0.20.0` as `someModule`
2. Refer to the module's outputs using `someModule.outputs.resourceId` and set it as a non-secure output variable in your own module, for example `output someId string = someModule.outputs.resourceId`

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.44.0

**Comment by BernieWhite** [comment]
@ThijmenDam Thanks for logging an issue.

Are you able to:

- Confirm or deny that the output is using `listOutputsWithSecureValues` when compiled to JSON with `bicep build`.
- Provide the Bicep CLI version you are using.

Based on my local testing of your case, I'm getting:

```
"outputs": {
  "storageAccountFunctionId": {
    "type": "string",
    "value": "[listOutputsWithSecureValues(resourceId('Microsoft.Resources/deployments', 'abc'), '2022-09-01').resourceId]"
  }
}
```

Which I think might be related to this bug in Bicep generation: https://github.com/Azure/bicep/issues/16991

I'm going to take some time to investigate it a little more to find out the specifics of `listOutputsWithSecureValues` and if it is safe to exclude, but normally `list*` are true positives.

**Comment by ThijmenDam** [comment]
Appreciate the quick response @BernieWhite.

> Confirm or deny that the output from your complete code is using `listOutputsWithSecureValues` when compiled to JSON with `bicep build`.

Confirmed:

```bicep
  "outputs": {
    "storageAccountFunctionId": {
      "type": "string",
      "value": "[listOutputsWithSecureValues(resourceId('Microsoft.Resources/deployments', format('{0}-{1}-{2}', parameters('identifier'), parameters('storageAccountName'), parameters('deploymentTimeStamp'))), '2022-09-01').resourceId]"
    }
  }
```

> Provide the Bicep CLI version you are using.

0.36.1

> Which I think might be related to this bug in Bicep generation: https://github.com/Azure/bicep/issues/16991

Agreed. Thanks for taking the time to look into this, would you let me know if I can be of any help?

**Comment by BernieWhite** [comment]
This has been released as Bicep CLI v0.36.177.
